tests/backup/backup_kubevirt_test.go

```diff
@@ -1540,8 +1540,223 @@ var _ = Describe("{KubevirtVMWithFreezeUnfreeze}", func() {
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of restore %s from backup %s", restoreName, backupName))
+		})
+
+		Step("Validating in place restore with retain on frozen VM", func() {
+			log.InfoD("Validating in place restore with retain on frozen VM")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+
+			expectedRestoredAppContexts := make([]*scheduler.Context, 0)
+			for _, scheduledAppContext := range scheduledAppContexts {
+				expectedRestoredAppContext, err := CloneAppContextAndTransformWithMappings(scheduledAppContext, make(map[string]string), make(map[string]string), true)
+				if err != nil {
+					log.Errorf("TransformAppContextWithMappings: %v", err)
+					continue
+				}
+				expectedRestoredAppContexts = append(expectedRestoredAppContexts, expectedRestoredAppContext)
+			}
+			err = ValidateRestore(ctx, restoreName, BackupOrgID, expectedRestoredAppContexts, make([]string, 0))
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Validating restore [%s] with retain", restoreName))
+		})
+
+		Step("Performing in-place restore with replace option", func() {
+			log.InfoD("Performing in-place restore with replace option")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			restoreName = fmt.Sprintf("%s-%s", "in-place-restore-replace", RandomString(6))
+			restoreNames = append(restoreNames, restoreName)
+			log.InfoD("Restoring the [%s] backup", backupName)
+			// Not restoring with validation as it will fail for all the VMs which are gone in scheduling state
+			err = CreateRestoreWithReplacePolicy(restoreName, backupName, make(map[string]string), DestinationClusterName, BackupOrgID, ctx, make(map[string]string), ReplacePolicyDelete)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of restore %s from backup %s", restoreName, backupName))
+		})
+
+		Step("Validating in place restore with replace", func() {
+			log.InfoD("Validating in place restore with replace")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+
+			expectedRestoredAppContexts := make([]*scheduler.Context, 0)
+			for _, scheduledAppContext := range scheduledAppContexts {
+				expectedRestoredAppContext, err := CloneAppContextAndTransformWithMappings(scheduledAppContext, make(map[string]string), make(map[string]string), true)
+				if err != nil {
+					log.Errorf("TransformAppContextWithMappings: %v", err)
+					continue
+				}
+				expectedRestoredAppContexts = append(expectedRestoredAppContexts, expectedRestoredAppContext)
+			}
+			err = ValidateRestore(ctx, restoreName, BackupOrgID, expectedRestoredAppContexts, make([]string, 0))
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Validating restore [%s] with replace", restoreName))
+		})
+
+	})
+
+	JustAfterEach(func() {
```

Comment:
Please delete backups and restores also as part of JustAfterEach as CR validations(to be added in future) will fail for this test.

Reply:
@atrivedi-px Backups and restores will be cleaned up as soon as the backup location is deleted and the cluster is deleted.

Reply:
CleanupCloudSettingsAndClusters has the validation added before backup/restore or cluster deletion https://github.com/portworx/torpedo/blob/9d3451fd1c7a5952459e5b0fc796643b26dc7d6e/tests/backup_helper.go#L1598

Reply:
Why don't we add it at the end? Is it because you will need the cluster reference?

Reply:
But we add the CR cleanup check right after deleting the backup location, then we can avoid backup deletion in the test case because backup location deletion will cleanup the backups as well.

Reply:
Yes @mkoppal-px , we needed cluster reference that's why added before cluster deletion, we can move it after cluster deletion also by using kubeconfigs instead of cluster references

Reply:
Lets discuss tomorrow before merging.